gdal/swig/python/osgeo/utils/gdal2xyz.py

```diff
@@ -78,9 +85,15 @@ def main(argv):
             band_nums.append(int(argv[i + 1]))
             i = i + 1
 
+        elif arg == '--allBands':
+            band_nums = None
```

Comment:
I find it a bit confusing that the semantics of [] or None is different. a allBands = True boolean would be more readable

Reply:
are you referring only to main() or also to gdal2xyz()?
I using another parameter for bands in gdal2xyz would be confusing.
Now that we use Python3, Another conversion that we could adopt is using Ellipsis (...) for "auto" or "default" value, i.e.        ```elif arg == '--allBands':
            band_nums = ...```

Reply:
let's forget my comment. I missed that the code was split in 2 functions. The code is OK as it

Reply:
thanks!

Reply:
@rouault I've addressed that issue you raised nonetheless.